docs/examples/clusters/glusterfs/userdata/bootstrap

```diff
@@ -0,0 +1,152 @@
+#!/bin/bash
+#yum update -y
+#######################################################################################################################################################
+### This bootstrap script does the following
+### 1- install gluster packages on all nodes 
+### 2- on glusterfs-server<X> nodes it formats the NVME disks (HighIO1.36 shape), creates a LVM LV called brick<X> (XFS)
+### 3- fixes the resolve.conf file. GlusterFS needs DNS to work properly so make sure you update the below domains to match your environment
+### 4- disable local firewall. Feel free to update this script to open only the required ports. I have a security list rule that allows communication
+### within my tenancy (subnet Source: 172.0.0.0/16  IP Protocol: All Protocols Allows: all traffic for all ports
+### 5- install and configure a gluster volume called glustervol1 using server1-brick1, server2-brick2 and server3-brick3 LVs (replicas)
+### 6- creates a local mount point in the glusterFS clients
+### 7- add a local entry in the clients Fstab with '_netdev' attribute. IMPORTANT to avoid boot issues.
+### 
+### Feel free to contact me in case of any questions - Gilson Melo - gilson.melo@oracle.com
+######################################################################################################################################################
+exec 2>/dev/null
+glusterfshost=$(hostname -s)
```

Comment:
I thought Chef/Puppet are the solutions that we recommend to people for setting up a particular instance with a particular set of libraries, applications etc. Does Chef etc. not work for this glusterFs installation, or am I mistaken in my understanding of Chef etc. ?

I just saw an email from Brian which described how Docker/Kubernetes are becoming industry standards. Looks like GlusterFs can be installed using docker - https://github.com/gluster/gluster-containers, and thereby orchestrated using k8s.

I am wondering if we should, at this point, start assuming our examples to be code that promotes the state-of-the-art. I will assume that people will copy-paste this code as is. Hence if we want our customers to adopt Chef/Docker etc., our examples should do that as well.

Reply:
Depends on the purpose of "docs/examples".  If the directory is for Oracle to show it's recommendations, you might be right. (but I would even argue if it is true because OCI provides an infrastructure where customers can use however they want.)  On the other hand, if the purpose is just to show a simple example, I think each example should focus on only its subject (in this case glusterfs).